tools/release_builder.py

```diff
@@ -205,394 +173,283 @@ def extract_json_file(package_dict, name):
         return pkgdir
 
 
-    def _download_unpack_stub_universe(self, scratchdir):
-        '''Returns the path to the package directory in the stub universe.'''
-        _, stub_universe_extension = os.path.splitext(self._stub_universe_url)
-        if stub_universe_extension == '.zip':
-            # stub universe zip package (universe 2.x only)
-            with urllib.request.urlopen(self._stub_universe_url) as stub_universe_file:
-                return self._unpack_stub_universe_zip(scratchdir, stub_universe_file)
-        elif stub_universe_extension == '.json':
-            # stub universe json file (universe 3.x+ only)
-            with urllib.request.urlopen(self._stub_universe_url) as stub_universe_file:
-                return self._unpack_stub_universe_json(scratchdir, stub_universe_file)
-        else:
-            raise Exception('Expected .zip or .json extension for stub universe: {}'.format(
-                self._stub_universe_url))
+    def _update_package_dot_json(self, package_json):
+        '''Updates the package.json definition to contain the desired version string,
+        and updates the package to reflect any beta or non-beta status as necessary.
+        '''
+        orig_package_json = package_json.copy()
 
+        # For beta releases, always clear 'selected'
+        if self._beta_release:
+            package_json['selected'] = False
 
-    def _update_file_content(self, path, orig_content, new_content, showdiff=True):
-        if orig_content == new_content:
-            logger.info('No changes detected in {}'.format(path))
-            # no-op
-        else:
-            if showdiff:
-                logger.info('Applied templating changes to {}:'.format(path))
-                logger.info('\n'.join(difflib.ndiff(orig_content.split('\n'), new_content.split('\n'))))
+        # Update package's name to reflect any changes due to BETA=on/off
+        package_json['name'] = self._pkg_name
+        # Update package's version to reflect the user's input
+        package_json['version'] = self._pkg_version
+        # Update package's upgradesFrom/downgradesTo to reflect any package name changes
+        # due to enabling or disabling a beta bit.
+        if self._stub_universe_pkg_name != self._pkg_name:
+            last_release = self._pkg_manager.get_latest(self._pkg_name)
```

Comment:
We may want to make this a requirement for certain packages? IIRC `get_latest` always returns something. For `hello-world` we also allow `*`.